File: windows-agent-csharp/src/KaderoAgent/Audit/AuditEvent.cs
namespace KaderoAgent.Audit;

public class AuditEvent
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string EventType { get; set; } = "";  // SESSION_LOCK, SESSION_UNLOCK, PROCESS_START, PROCESS_STOP, SESSION_LOGON, SESSION_LOGOFF
    public DateTime EventTs { get; set; } = DateTime.UtcNow;
    public string? SessionId { get; set; }
    public Dictionary<string, object> Details { get; set; } = new();
}

File: web-dashboard/src/types/audit-event.ts
export type AuditEventType =
  | 'SESSION_LOCK'
  | 'SESSION_UNLOCK'
  | 'SESSION_LOGON'
  | 'SESSION_LOGOFF'
  | 'PROCESS_START'
  | 'PROCESS_STOP';

export interface AuditEvent {
  id: string;
  event_type: AuditEventType;
  event_ts: string; // ISO 8601
  session_id: string | null;
  details: {
    process_name?: string;
    pid?: number;
    window_title?: string;
    exe_path?: string;
    reason?: string;
  };
}

export interface DeviceAuditEventsResponse {
  device_id: string;
  date: string;
  timezone: string;
  total_elements: number;
  events: AuditEvent[];
}
